src/librustc/hir/lowering.rs

```diff
@@ -2997,7 +2997,10 @@ impl<'a> LoweringContext<'a> {
                 let mut body = body.clone();
 
                 for a in arguments.iter().rev() {
```

Comment:
It would be nice to have a comment laying out the "overall strategy" here -- something [roughly like this zulip comment](https://rust-lang.zulipchat.com/#narrow/stream/187312-t-compiler.2Fwg-async-await/topic/.2354716.20drop.20order/near/164610870).

Reply:
Fixed.